frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -1255,6 +1342,102 @@ pub mod pallet {
                 && latest_staked_value > Zero::zero()
         }
 
+        /// return the delegatee_address if exists to the max_depth
+        /// else return the delegator address if no delegatee exists
+        pub(crate) fn check_delegate(
+            delegator_address: T::AccountId,
+            contract_id: T::SmartContract,
+        ) -> Result<T::AccountId,DispatchError> {
+
+            // delegatee account must be set in order to claim rewards
+            ensure!(DelegateInfo::<T>::contains_key(&delegator_address, &contract_id), Error::<T>::DelegateeAccountNotSetForContract);
+
+            let mut previous_address: T::AccountId = delegator_address;
+
+            // return delegatee address following the mapping in DelegateInfo upto MAX_DEPTH_FOR_DELEGATION
+            for _ in 0..MAX_DEPTH_FOR_DELEGATION {
+                if let Ok(delegatee_address) =
+                    DelegateInfo::<T>::try_get(&previous_address, &contract_id)
+                {
+                    previous_address = delegatee_address;
+                } else {
+                    return Ok(previous_address);
+                }
+            }
+
+            return Ok(previous_address);
+        }
+
+        /// add delegatee address for a caller for a given contract_id
+        pub fn set_delegatee_info(
+            delegator_address: T::AccountId, 
+            contract_id: T::SmartContract, 
+            delegatee_address: T::AccountId
+        ) -> Result<(),DispatchError> {
+
+            // check of delegating to self
+            ensure!(
+                delegator_address != delegatee_address,
+                Error::<T>::SelfDelegationNotAllowed
+            );            
+
+            //check if contract is registered dapp
+            ensure!(
+                RegisteredDapps::<T>::contains_key(&contract_id),
+                Error::<T>::NotOperatedContract,
+            );
```

Comment:
Question - if contract has been unregistered, does is still make sense to allow delegating rewards for it? And why?

Reply:
I think it depends on how much power do we want to give to stakers. 
Even if the contract is unregistered, staker may have some funds left to be claimed and they might want it to be withdrawn to another account, if they wish to accomplish such a thing then they can change the delegatee_account to their preferred account.

I personally think this should not be allowed once the contract is unregistered and if staker wish their rewards to be transferred to another account then they should do it manually.
But it is no big a deal, we can allow it if we want.